core/server/services/routing/helpers/context.js

```diff
@@ -18,7 +18,8 @@ const labs = require('../../labs'),
     subscribePattern = new RegExp('^\\/subscribe\\/'),
     // routeKeywords.amp: 'amp'
     ampPattern = new RegExp('\\/amp\\/$'),
-    homePattern = new RegExp('^\\/$');
+    homePattern = new RegExp('^\\/$'),
+    previewPattern = new RegExp('^\\/p\\/');
```

Comment:
This is a great PR! I didn't even know that the preview pages were being indexed under the described conditions. I think it makes sense to add `<meta name="robots" content="noindex,nofollow" />` to all preview pages using the following RegExp:

```
\/p\/
```
RegEx Tester for the solution above: https://regex101.com/r/jRBDSJ/1
RegEx Tester for [current PR](https://github.com/TryGhost/Ghost/pull/9762/files#diff-e7b88060df0e3026e0ea24c3b5b4671cR22): https://regex101.com/r/5CqTPx/1

Reply:
Thanks for the suggestion! This regex would actually catch all instances of preview pages, since the regex is executed on the  [relative url](https://github.com/vikaspotluri123/Ghost/blob/8ca88cebeee4000fc7ca3d7165eae6747c130af0/core/server/services/routing/helpers/context.js#L50). This still allows the client to use, for example, a tag w/ slug `p` (i.e. `my.blog/tag/p/` would still match w/o the line start operater)

You can see similar / the same logic with the `subscribePattern` test 😄

Reply:
Thanks for clarifying the logic behind the relative URLs, and where to find them in the code base. I love using Ghost! 👻